src/Utility.js

```diff
@@ -88,24 +88,8 @@ export var promise = typeof Promise === 'function' ? Promise : function (callbac
  * @param {function} callback
  * @return {object}
  */
-export function request (value) {
-	return new promise(function (resolve) { animation(function () { resolve(value) }) })
-}
-
-/**
- * @param {function} callback
- * @return {object}
- */
-export function respond (callback) {
-	return request(callback).then(callback)
-}
-
-/**
- * @param {function} callback
- * @return {number}
- */
-export function animation (callback) {
-	return typeof requestAnimationFrame === 'function' ? requestAnimationFrame(callback) : setTimeout(callback, 16)
+export function respond (value) {
+	return new promise(function (resolve) { timeout(function () { resolve(value) }, 16) })
```

Comment:
I don't understand this. Why remove `requestAnimationFrame`?

Reply:
setTimeout is more deterministic and after paint.

Reply:
1. `setTimeout` does not support high refresh rate (e.g. 144Hz) monitors
2. we want before paint execution, not after paint

Reply:
Before paint execution, why?

Reply:
The purpose of `requestAnimationFrame` is to allow many v-tree updates to occur within a single frame without incurring the DOM interface penalty. That is, delay actually writing to the DOM as long as possible (i.e. `requestAnimationFrame`) so that you can 'batch' (and de-dupe) DOM mutation.

If you delay until after paint then you lose a frame.

Reply:
I see. So you content with React's idea of delegating effects till after paint – can that have any observable effects between React and Dyo?

Reply:
I guess the main effect would be that you lose a frame. I'd say that's 'observable'
e.g.
```coffee
Slow = ->
  [value, setValue] = useState 0
  useEffect ->
    heights = 0
    for i in [0..100]
      heights += $el.getBoundingClientRect().height
    setValue heights
  
  h 'div', "initial heights: #{value}"
```

If `useEffect` takes too long to run (and because you delayed execution all the way until `requestAnimationFrame` instead of letting it compute as soon as possible), then you miss the frame boundary.

The `useLayout` version would be 1 frame faster.

```coffee
Fast = ->
  useLayout ->
    Promise.resolve().then ->
       # expensive computation, after paint but before requestAnimationFrame
```